dataplane/api-server/src/config.rs

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+*/
+use clap::Parser;
+use std::path::PathBuf;
+
+#[derive(Debug, Parser, Clone)]
+pub struct GrpcConfig {
+    #[clap(long, default_value = "false")]
+    pub enable_tls: bool,
+    #[clap(long, default_value = "false")]
+    pub enable_mtls: bool,
+    pub certificate_authority_root_path: Option<PathBuf>,
+    pub server_certificate_path: Option<PathBuf>,
+    pub server_private_key_path: Option<PathBuf>,
+    pub client_certificate_authority_root_path: Option<PathBuf>,
+    pub client_certificate_path: Option<PathBuf>,
+    pub client_private_key_path: Option<PathBuf>,
+}
+
+impl GrpcConfig {
+    pub fn validate(&self) -> Result<(), String> {
+        fn validate_paths(paths: &[(&Option<PathBuf>, &str)]) -> Result<(), String> {
```

Comment:
is this solely done to avoid a move? asking because i'm not a big fan of inline functions unless they're essential. this can be just its own function that accepts a vector. please correct me if i'm wrong.

Reply:
No moving at play here. I generally agree nested functions suck and avoid them. In this case, I felt it wasn't that big of a deal because it's not intended to be used anywhere outside of the validate method. I tend to feel extremely tight coupling is the only excuse for nested functions.

Either way, I moved it out.